api/ResourceApi.cpp

```diff
@@ -0,0 +1,78 @@
+#include "ResourceApi.h"
+
+#include "constants.hpp"
+
+#include <spdlog/spdlog.h>
+
+namespace io {
+namespace swagger {
+namespace server {
+namespace api {
+
+using namespace io::swagger::server::model;
+
+ResourceApi::ResourceApi(Pistache::Address addr)
+    : httpEndpoint(std::make_shared<Pistache::Http::Endpoint>(addr))
+{
+}
+
+void ResourceApi::init(size_t thr = 2) {
+    auto opts = Pistache::Http::Endpoint::options()
+        .threads(thr);
+    httpEndpoint->init(opts);
+    setupRoutes();
+}
+
+void ResourceApi::start() {
+    httpEndpoint->setHandler(router.handler());
+    httpEndpoint->serve();
+}
+
+void ResourceApi::shutdown() {
+    httpEndpoint->shutdown();
+}
+
+void ResourceApi::setupRoutes() {
+    using namespace Pistache::Rest;
+
+    Routes::Post(router, irods::rest::base_url + "/resource", Routes::bind(&ResourceApi::repl_handler, this));
+    Routes::Delete(router, irods::rest::base_url + "/resource", Routes::bind(&ResourceApi::trim_handler, this));
```

Comment:
Just for informational purposes, the route that these live behind will eventually be something like `/logicalpath/[trim | repl]`